subsys/net/lib/mqtt_socket/mqtt_encoder.c

```diff
@@ -0,0 +1,781 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt_encoder.c
+ *
+ * @brief Encoding functions needed to create packet to be sent to the broker.
+ */
+
+#include "mqtt_internal.h"
+#include "mqtt_os.h"
+
+#define MQTT_3_1_0_PROTO_DESC_LEN 6
+#define MQTT_3_1_1_PROTO_DESC_LEN 4
+
+const u8_t mqtt_3_1_0_proto_desc_str[MQTT_3_1_0_PROTO_DESC_LEN] = {
```

Comment:
Could this be static as I could not find any access to it outside of this source file?